modules/authglobus/lib/Auth/Source/Globus.php

```diff
@@ -0,0 +1,273 @@
+    *
+    * @param string $name
+    *
+    * @return an array containing first name and last name
+    */
+    protected function getFullName($name) {
+        $trimmedName = trim($name);
+
+        if ($trimmedName && (strpos($trimmedName, ',') !== null || strpos($trimmedName, '\s') !== null)) {
+            $delimiter = strpos($name, ',') !== null ? ',' : '\s';
+            $nameParts = explode($delimiter, $trimmedName, 2);
+            $nameParts = array_map('trim', $nameParts);
+
+            $firstName;
+            $lastName;
+
+            if ($delimiter === '\s') {
+                $firstName = $nameParts[0];
+                $lastName = $nameParts[1];
+            } else {
+                $firstName = $nameParts[1];
+                $lastName = $nameParts[0];
+            }
+
+            return array(
+                'first_name' => $firstName,
+                'last_name' => $lastName
+            );
+        }
+
+        return array(
+            'first_name' => 'UNKNOWN',
+            'last_name' => 'UNKNOWN'
+        );
+    }
+
+    /**
+    * Wrapper for Curl Requests
+    *
+    * @param string path
+    * @param string signature
+    * @param string token
+    * @param string qs
+    * @param array contents
+    * @param string method
+    *
+    * @return The response from the endpoint where we made the request.
+    */
+    protected function doCurlRequest(
+        $path,
+        $signature = null,
+        $token = null,
+        $qs = null,
+        $contents = null,
+        $method = 'GET'
+    ) {
+        curl_setopt($this->curl, CURLOPT_USERAGENT, 'SSPHP Globus'); //umad Globus
+
+        if ($signature !== null) {
+            curl_setopt($this->curl, CURLOPT_USERPWD, $signature);
+        }
+        
+        curl_setopt($this->curl, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($this->curl, CURLOPT_FOLLOWLOCATION, true);
+        curl_setopt($this->curl, CURLOPT_RETURNTRANSFER, 1);
+
+        curl_setopt($this->curl, CURLOPT_SSL_VERIFYHOST, false);
+        curl_setopt($this->curl, CURLOPT_SSL_VERIFYPEER, false);
+        curl_setopt($this->curl, CURLINFO_HEADER_OUT, true);
+
+        $endPoint = self::API_ENDPT . $path . (($qs !== null) ? $qs : '');
+        curl_setopt($this->curl, CURLOPT_URL, $endPoint);
+
+        if ($method === 'POST') {
+            curl_setopt($this->curl, CURLOPT_POSTFIELDS, OauthUtil::build_http_query($contents));
+        } else { // do GET by default
+            $header = 'Authorization: Bearer ' . $token;
+            curl_setopt($this->curl, CURLOPT_HTTPHEADER, array($header));
+            curl_setopt($this->curl, CURLOPT_HTTPGET, true);
+        }
+
+        $result = curl_exec($this->curl);
+        $outInfo = curl_getinfo($this->curl);
+
+        if (curl_error($this->curl)) {
+            throw new SimpleSAML_Error_Error($outInfo['http_code'] . ': Could not successfully reach Globus service.');
```

Comment:
is there actually always going to be an http_code on this?

Seems like this could be a connection timeout that might through this curl error, it doesnt seem to be specifically for HTTP errors (https://curl.haxx.se/libcurl/c/libcurl-errors.html)